sim/core/character.go

```diff
@@ -338,45 +340,31 @@ func (character *Character) AddPet(pet PetAgent) {
 		panic("Pets must be added during construction!")
 	}
 
-	character.Pets = append(character.Pets, pet)
-}
-
-func (character *Character) GetPet(name string) PetAgent {
-	for _, petAgent := range character.Pets {
-		if petAgent.GetPet().Name == name {
-			return petAgent
-		}
-	}
-	panic(character.Name + " has no pet with name " + name)
+	character.PetAgents = append(character.PetAgents, pet)
+	character.Pets = append(character.Pets, pet.GetPet())
 }
 
 func (character *Character) MultiplyMeleeSpeed(sim *Simulation, amount float64) {
 	character.Unit.MultiplyMeleeSpeed(sim, amount)
 
-	if len(character.Pets) > 0 {
```

Comment:
Have you checked perf changes on this, or at least an overall perf comparison for this PR?

Reply:
The len() checks before iterating have no influence on performance, at least with go 1.21. Overall, this is a 3-4% perf increase for dual-wielding classes (go 1.21, ARM64)

Reply:
Just retested for BM hunter: that's an even larger increase, 6-7%. Virtual calls are remarkably "expensive", and most of them are gone now for pets.